zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/ListViewFlowNodeFromProcesInstanceHandlerTest.java

```diff
@@ -0,0 +1,289 @@
+        .isEqualTo(processInstanceRecordValue.getProcessInstanceKey());
+    assertThat(flowNodeInstanceForListViewEntity.getTenantId())
+        .isEqualTo(processInstanceRecordValue.getTenantId());
+    assertThat(flowNodeInstanceForListViewEntity.getPartitionId())
+        .isEqualTo(processInstanceRecord.getPartitionId());
+    assertThat(flowNodeInstanceForListViewEntity.getPosition())
+        .isEqualTo(processInstanceRecord.getPosition());
+    assertThat(flowNodeInstanceForListViewEntity.getActivityId())
+        .isEqualTo(processInstanceRecordValue.getElementId());
+    assertThat(flowNodeInstanceForListViewEntity.getActivityState())
+        .isEqualTo(FlowNodeState.ACTIVE);
+    assertThat(flowNodeInstanceForListViewEntity.getJoinRelation())
+        .isEqualTo(
+            new ListViewJoinRelation(ListViewTemplate.ACTIVITIES_JOIN_RELATION)
+                .setParent(processInstanceRecordValue.getProcessInstanceKey()));
+  }
+
+  @Test
+  void shouldUpdateStateForCompletedRecord() {
+    // given
+    final long timestamp = new Date().getTime();
+    final ProcessInstanceRecordValue processInstanceRecordValue =
+        ImmutableProcessInstanceRecordValue.builder()
+            .from(factory.generateObject(ProcessInstanceRecordValue.class))
+            .withBpmnElementType(BpmnElementType.SERVICE_TASK)
+            .build();
+    final Record<ProcessInstanceRecordValue> processInstanceRecord =
+        factory.generateRecord(
+            ValueType.PROCESS_INSTANCE,
+            r ->
+                r.withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED)
+                    .withTimestamp(timestamp)
+                    .withValue(processInstanceRecordValue));
+
+    // when then
+    final FlowNodeInstanceForListViewEntity entity = new FlowNodeInstanceForListViewEntity();
+    underTest.updateEntity(processInstanceRecord, entity);
+
+    assertThat(entity.getActivityState()).isEqualTo(FlowNodeState.COMPLETED);
+  }
+
+  @Test
+  void shouldUpdateEndTimeForTerminatedRecord() {
+    // given
+    final long timestamp = new Date().getTime();
+    final ProcessInstanceRecordValue processInstanceRecordValue =
+        ImmutableProcessInstanceRecordValue.builder()
+            .from(factory.generateObject(ProcessInstanceRecordValue.class))
+            .withBpmnElementType(BpmnElementType.SERVICE_TASK)
+            .build();
+    final Record<ProcessInstanceRecordValue> processInstanceRecord =
+        factory.generateRecord(
+            ValueType.PROCESS_INSTANCE,
+            r ->
+                r.withIntent(ProcessInstanceIntent.ELEMENT_TERMINATED)
+                    .withTimestamp(timestamp)
+                    .withValue(processInstanceRecordValue));
+
+    // when then
+    final FlowNodeInstanceForListViewEntity entity = new FlowNodeInstanceForListViewEntity();
+    underTest.updateEntity(processInstanceRecord, entity);
+
+    assertThat(entity.getActivityState()).isEqualTo(FlowNodeState.TERMINATED);
```

Comment:
An assertion for the `endTime` should be required here? Same applies for the case above.

Reply:
No, here not. When storing flow-node-instance in `list-view`, we store the shortened version of object without end or start date.

Reply:
Could you rename the test in that case? :pray:

Reply:
Haha, sure :) Missed that)